db/migrate/20200106134953_add_unique_index_to_benchmarks.rb

```diff
@@ -0,0 +1,48 @@
+class AddUniqueIndexToBenchmarks < ActiveRecord::Migration[5.2]
+  def change
+    Xccdf::Benchmark.transaction do
+      benchmarks = {}
+
+      Xccdf::Benchmark.includes(:rules, :profiles).find_each do |bm|
+        if benchmarks[[bm.ref_id, bm.version]]
+          migrate_rules(benchmarks[[bm.ref_id, bm.version]], bm)
+          migrate_profiles(benchmarks[[bm.ref_id, bm.version]], bm)
+          bm.destroy! # profiles, rules
+        else
+          benchmarks[[bm.ref_id, bm.version]] = bm
+        end
+      end
+    end
```

Comment:
Can we get this block out into a service? It would allow to get some tests to ensure this works on all scenarios and ensure we don't break this with future commits before it gets deployed to prod.